Tests/WorkspaceTests/WorkspaceTests.swift

```diff
@@ -6502,6 +6502,131 @@ final class WorkspaceTests: XCTestCase {
         }
     }
 
+    func testArtifactDownloadArchiveExists() throws {
+        let sandbox = AbsolutePath("/tmp/ws/")
+        let fs = InMemoryFileSystem()
+        // this relies on internal knowledge of the destination path construction
```

Comment:
This seems fine for a unit test.  We could always extract into a static property for the subpath if we feel it's important enough.